experimental/Schemes/Tjurina.jl

```diff
@@ -0,0 +1,770 @@
+    of multivariate polynomial ring in 2 variables x, y
+      over prime field of characteristic 2
+    by ideal (x^3 + y^2, x^2, 0)
+  at complement of maximal ideal of point (0, 0)
+
+julia> tjurina_algebra(L(x^3-y^2), 1)
+Localization
+  of quotient
+    of multivariate polynomial ring in 2 variables x, y
+      over prime field of characteristic 2
+    by ideal (x^3 + y^2, x^3, 0, x^2*y, 0)
+  at complement of maximal ideal of point (0, 0)
+```
+"""
+function tjurina_algebra(f::MPolyLocRingElem{<:Any, <:Any, <:Any, <:Any, <:MPolyComplementOfKPointIdeal}, k::Integer = 0)
+  X = HypersurfaceGerm(quo(parent(f), ideal(parent(f), f))[1])
+  return tjurina_algebra(X, k)
+end
+
+
+
+
+
+###############################################################################
+
+#####                           Tjurina number                            #####
+
+###############################################################################
+
+@doc raw"""
+    tjurina_number(f::MPolyRingElem)
+
+Return the global Tjurina number of a polynomial `f`. Return -1 if infinite.
+# Examples
+```jldoctest
+julia> R,(x,y) = QQ["x", "y"];
+
+julia> f = x*(x-1)*y;
+
+julia> tjurina_number(f)
+2
+```
+"""
+function tjurina_number(f::MPolyRingElem)
+  isa(coefficient_ring(f), AbstractAlgebra.Field) || error("The polynomial requires a coefficient ring that is a field.")
+  return vector_space_dimension(tjurina_algebra(f))
+end
+
+@doc raw"""
+    tjurina_number(X::AffineScheme{<:Field,<:MPolyQuoRing})
+
+Return the global Tjurina number of the affine scheme `X`, if `X` is a hypersurface.
+Return -1 if infinite.
+# Examples
+```jldoctest
+julia> R,(x,y) = QQ["x", "y"];
+
+julia> f = x^3 - y^2;
+
+julia> X = AffineScheme(quo(R, ideal(R, x^3-y^2))[1])
+Spectrum
+  of quotient
+    of multivariate polynomial ring in 2 variables x, y
+      over rational field
+    by ideal (x^3 - y^2)
+
+julia> tjurina_number(X)
+2
+```
+"""
+function tjurina_number(X::AffineScheme{<:Field,<:MPolyQuoRing}) 
+  return vector_space_dimension(tjurina_algebra(X))
+end
+
+
+
+@doc raw"""
+    tjurina_number(X::HypersurfaceGerm)
+    tjurina_number(X::HypersurfaceGerm, k::Integer = 0)
+
+Return the `k`-th local Tjurina number of `(X,p)` at `p`. 
+By default computes the local Tjurina number (`k=0`) at `p`.
+Higher Tjurina numbers are of interest in positive characteristic. Return -1 if infinite.
+# Examples
+```jldoctest
+julia> R,(x,y) = QQ["x", "y"];
+
+julia> f = x^3 - y^2;
+
+julia> X = HypersurfaceGerm(AffineScheme(quo(R, ideal(R, f))[1]), [0, 0]);
+
+julia> tjurina_number(X)
+2
+```
+"""
+function tjurina_number(X::HypersurfaceGerm, k::Integer = 0)                                    #TODO: simplify after vec_dim is improved
+  # Fix for zero dimensional vector space
+  I = modulus(tjurina_algebra(X, k))
+  if base_ring(I)(1) in I
+    return 0
+  end
+  # Fix for infinite dimensional vector space
+  try
+    return vector_space_dimension(tjurina_algebra(X, k))
+  catch
+    return -1
+  end
```

Comment:
You could test for the dimension of the Tjurina algebra and return -1, if it exceeds zero.

Reply:
This kind of test is problematic because it will intercept *any* error, not just one that indicates something about the dimension.

In general, it would be preferable to not handle this kind of thing via exceptions anyway.

BTW, why does the method just before it not need this "fix" (even though its docstring also says it can return `-1` to indicate infinity)?